src/components/TodoList.tsx

```diff
@@ -0,0 +1,98 @@
+import { useState } from "react";
+import { useTodoStore, ITodo } from "../useTodoStore";
+import { arrayMove } from "@dnd-kit/sortable";
+import {
+  DndContext,
+  closestCenter,
+  DragEndEvent,
+  DragOverlay,
+  DragStartEvent,
+} from "@dnd-kit/core";
+import Droppable from "./Droppable";
+import TodoItem from "./TodoItem";
+import styled from "styled-components";
+
+const TodoLists = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+function TodoList() {
+  const todos = useTodoStore((state) => state.todos); // 기존 todo list 배열
+  const updateTodos = useTodoStore((state) => state.updateTodos); // todo list 순서 업데이트 함수
+  const updateCategory = useTodoStore((state) => state.updateCategory); // 카테고리 변경 시 todo list 순서 업데이트 함수
+  const [activeTodo, setActiveTodo] = useState<ITodo | null>(null);
+
+  const lists = [
+    {
+      id: "todo",
+      title: "To Do",
+      items: todos.filter((todo) => todo.category === "todo"),
+    },
+    {
+      id: "doing",
+      title: "Doing",
+      items: todos.filter((todo) => todo.category === "doing"),
+    },
+    {
+      id: "done",
+      title: "Done",
+      items: todos.filter((todo) => todo.category === "done"),
+    },
+  ];
+
+  const onDragStart = (event: DragStartEvent) => {
+    const { active } = event;
+    setActiveTodo(todos.find((todo) => todo.id === active.id) || null);
+  };
+  const onDragEnd = (event: DragEndEvent) => {
+    setActiveTodo(null);
+    const { active, over } = event;
+    if (!over || active.id === over.id) return;
+
+    // 1. 드롭한 위치의 새로운 카테고리 찾기
+    let newCategory = over?.data.current?.sortable?.containerId || over.id;
```

Comment:
```suggestion
    if (!over || active.id === over.id) return;

    // 1. 드롭한 위치의 새로운 카테고리 찾기
    const newCategory = over.data.current?.sortable?.containerId || over.id;
```
- newCategory가 변경되는 값이 아니면 const로 선언해주는게 좋습니다
- 위에서 !over일 때 early return 시키고 있기 때문에, over는 truthy한 값임이 보장되어서 옵셔널체이닝을 사용하지 않아도 됩니다